### PR TITLE
Merge branch release/9.0 into main

### PR DESCRIPTION
**Pull request by joperezr**
This should be merged and not squashed.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/extensions/pull/5632)

### CODE REVIEW

File: src/Libraries/Microsoft.Extensions.AI.Abstractions/Microsoft.Extensions.AI.Abstractions.csproj [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
   <PropertyGroup>
     <Stage>preview</Stage>
+    <SuppressFinalPackageVersion>true</SuppressFinalPackageVersion>
```
Comment (stephentoub) [inline src/Libraries/Microsoft.Extensions.AI.Abstractions/Microsoft.Extensions.AI.Abstractions.csproj:11]: What does this do?

**Review by stephentoub — APPROVED** [review]

File: src/Libraries/Microsoft.Extensions.AI.Abstractions/Microsoft.Extensions.AI.Abstractions.csproj (hunk 1 above)
Comment (joperezr) [reply src/Libraries/Microsoft.Extensions.AI.Abstractions/Microsoft.Extensions.AI.Abstractions.csproj:11]: Makes it such that the package version won't be stabilized even when the rest of the repo is going stable.

To pre-answer your follow-up question 😋, Yes ideally that was what setting Stage to preview should do, but I found that wasn't working and so this is the change that was made in the release branch in the meantime to make sure packages wouldn't ship stable. Now that this code will be in main and we are no longer on a time crunch, I'll investigate and remove the duplication after this goes in.

File: eng/Version.Details.xml [hunk 2]
```diff
@@ -1,188 +1,188 @@
...
+      <Sha>0456c7e91c34003f26acf8606ba9d20e29f518bd</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.NETCore.App.Ref" Version="9.0.0-rc.2.24473.5" CoherentParentDependency="Microsoft.AspNetCore.App.Runtime.win-x64">
+    <Dependency Name="Microsoft.NETCore.App.Ref" Version="9.0.0" CoherentParentDependency="Microsoft.AspNetCore.App.Runtime.win-x64">
       <Uri>https://dev.azure.com/dnceng/internal/_git/dotnet-runtime</Uri>
-      <Sha>990ebf52fc408ca45929fd176d2740675a67fab8</Sha>
+      <Sha>0456c7e91c34003f26acf8606ba9d20e29f518bd</Sha>
     </Dependency>
-    <Dependency Name="System.Numerics.Tensors" Version="9.0.0-rc.2.24473.5" CoherentParentDependency="Microsoft.AspNetCore.App.Runtime.win-x64">
+    <Dependency Name="System.Numerics.Tensors" Version="9.0.0">
```
Comment (RussKie) [inline eng/Version.Details.xml:83]: Why the change of the coherency dependency?

File: src/Libraries/Microsoft.Extensions.AI.Abstractions/Microsoft.Extensions.AI.Abstractions.csproj (hunk 1 above)
Comment (RussKie) [reply src/Libraries/Microsoft.Extensions.AI.Abstractions/Microsoft.Extensions.AI.Abstractions.csproj:11]: Shouldn't we set it in ProjectStaging.targets - a single point instead of changing multiple files?

```diff
diff --git a/eng/MSBuild/ProjectStaging.targets b/eng/MSBuild/ProjectStaging.targets
index e3a89d0354..6db968b7fb 100644
--- a/eng/MSBuild/ProjectStaging.targets
+++ b/eng/MSBuild/ProjectStaging.targets
@@ -28,4 +28,10 @@
     <Description Condition="'$(Stage)' == 'obsolete'">Obsolete Package. $(Description)</Description>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <SuppressFinalPackageVersion>false</SuppressFinalPackageVersion>
+    <!-- Makes it such that the package version won't be stabilized even when the rest of the repo is going stable. -->
+    <SuppressFinalPackageVersion Condition="'$(Stage)' == 'dev' or '$(Stage)' == 'preview'">true</SuppressFinalPackageVersion>
+  </PropertyGroup>
+
 </Project>
```

File: eng/Version.Details.xml (hunk 2 above)
Comment (joperezr) [reply eng/Version.Details.xml:83]: The coherent dependency was wrong for this item. When updating the dependencies internally I was getting an error mentioning it wasn't correct.
